src/main/webapp/user-location.js

```diff
@@ -0,0 +1,60 @@
+const apiKey = 'AIzaSyB5cba6r-suEYL-0E_nRQfXDtT4XW0WxbQ';
+const fetchJson = {
+  'considerIp': 'true',
+};
+const geoCodingUrl = 'https://maps.googleapis.com/maps/api/geocode/json?latlng=';
+const geoLocationUrl = 'https://www.googleapis.com/geolocation/v1/geolocate?key=';
+
+// Given the user's lat and lng from the geoLocation API,
+// getUserState calls to the geoCoding API to reverse geoCode
+// the coordinates and get the user's current state.
+async function getUserState(lat, lng) {
+  const geoCodingFetch = geoCodingUrl + lat + ',' + lng + '&key=' + apiKey;
+  return fetch(geoCodingFetch).then(function(response) {
```

Comment:
This and some other portions can be shortened using [arrow syntax](https://www.w3schools.com/js/js_arrow_function.asp) ie. .then((response) => response.json())

Reply:
Fixed!